pythonforandroid/toolchain.py

```diff
@@ -69,7 +78,7 @@ def format(self, record):
                                     # handler and reset the level
     logger.setLevel(logging.INFO)
     logger.touched = True
-    ch = logging.StreamHandler(stdout)
+    ch = logging.StreamHandler(stdout) if sys.stdout.isatty() else logging.NullHandler()
```

Comment:
if stdout is not a tty, it is not connected to a terminal. which (presumably) means it is actually piped into another script.  logging interspersed with actual data is ok for humans, it is very inconvenient for scripts.  If the logger actually wrote to a file, then I would have kept it, but since it only writes to stdout that seems neither wise nor convenient.  If logging to a file is subsequently added, then I'd be happy to revisit the logic.